via/services/ms_one_drive.py

```diff
@@ -0,0 +1,16 @@
+import re
+
+
+class MSOneDriveService:
+    URL_REGEXPS = (
+        re.compile(r"^https://.*\.sharepoint.com/.*download=1"),
+        re.compile(r"^https://api.onedrive.com/v1.0/.*/root/content"),
+    )
+
+    @classmethod
+    def is_one_drive_url(cls, url):
```

Comment:
This doesn't really need to be a service + class + method but for consistency when for the service.